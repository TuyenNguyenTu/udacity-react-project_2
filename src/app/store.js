import { configureStore } from '@reduxjs/toolkit';
import authedSlice from '../features/authSlice'
import questionSlice from '../features/questionSlice'
import userSlice from '../features/userSlice'
import loadingSlice from '../features/loadingSlice'

export const store = configureStore({
  reducer: {
    auth: authedSlice,
    question: questionSlice,
    user: userSlice,
    loading: loadingSlice
  },
});
