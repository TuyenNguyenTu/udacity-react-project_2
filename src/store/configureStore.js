import logger from "redux-logger";
import rootReducer from "./reducers";
import {  configureStore } from "@reduxjs/toolkit";


let store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false }).concat(logger)
});

export default store; 
