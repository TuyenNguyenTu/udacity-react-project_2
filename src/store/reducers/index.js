import { combineReducers } from "@reduxjs/toolkit";
import authenticateReducer from "./authenticateReducer";
import questionReducer from "./questionsReducer";
import usersReducer from "./usersReducer";

const rootReducer = combineReducers({
  auth: authenticateReducer,
  questions: questionReducer,
  users: usersReducer,
});

export default rootReducer;
