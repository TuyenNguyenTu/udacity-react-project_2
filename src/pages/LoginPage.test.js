import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store";
import Login from "./LoginPage";

describe("NewQuestionItem Test", () => {
  test("should render the component", () => {
    const view = render(
      <BrowserRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </BrowserRouter>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  test("should render correct data", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </BrowserRouter>
    );

    const UserNameInputElement = screen.getByTestId("Username");
    const PasswordInputElement = screen.getByTestId("Password");

    fireEvent.change(UserNameInputElement, { target: { value: "Username" } });
    fireEvent.change(PasswordInputElement, {
      target: { value: "Password" },
    });
    expect(UserNameInputElement.value).toBe("Username");
    expect(PasswordInputElement.value).toBe("Password");
  });

  test("should display message when user name incorrect", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </BrowserRouter>
    );

    const UserNameInputElement = screen.getByTestId("Username");
    const PasswordInputElement = screen.getByTestId("Password");
    const submitButtonElement = screen.getByTestId("buttonSubmit");
    const errorMessageElement = screen.getByTestId("errorMessage");
    fireEvent.change(UserNameInputElement, {
      target: { value: "" },
    });
    fireEvent.change(PasswordInputElement, {
      target: { value: "Password" },
    });
    fireEvent.click(submitButtonElement);
    expect(UserNameInputElement.value).toBe("");
    expect(PasswordInputElement.value).toBe("Password");
    setTimeout(() => {
      expect(errorMessageElement.value).toBe("username cannot empty");
    });
  });
});
