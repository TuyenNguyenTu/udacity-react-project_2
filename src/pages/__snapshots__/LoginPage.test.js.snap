// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestionItem Test should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-signin w-100 m-auto"
      >
        <form>
          <h1
            class="h3 mb-3 fw-normal"
          >
            Please sign in
          </h1>
          <small
            class="text-danger"
            data-testid="errorMessage"
          />
          <div
            class="form-floating"
          >
            <input
              class="form-control"
              data-testid="Username"
              placeholder="Username"
              type="text"
            />
            <label>
              Username
            </label>
          </div>
          <div
            class="form-floating "
          >
            <input
              class="form-control"
              data-testid="Password"
              placeholder="Password"
              type="password"
            />
            <label>
              Password
            </label>
          </div>
          <button
            class="btn btn-primary w-100 py-2"
            data-testid="buttonSubmit"
            type="button"
          >
            Sign in
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-signin w-100 m-auto"
    >
      <form>
        <h1
          class="h3 mb-3 fw-normal"
        >
          Please sign in
        </h1>
        <small
          class="text-danger"
          data-testid="errorMessage"
        />
        <div
          class="form-floating"
        >
          <input
            class="form-control"
            data-testid="Username"
            placeholder="Username"
            type="text"
          />
          <label>
            Username
          </label>
        </div>
        <div
          class="form-floating "
        >
          <input
            class="form-control"
            data-testid="Password"
            placeholder="Password"
            type="password"
          />
          <label>
            Password
          </label>
        </div>
        <button
          class="btn btn-primary w-100 py-2"
          data-testid="buttonSubmit"
          type="button"
        >
          Sign in
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
